UI/window-basic-status-bar.cpp

```diff
@@ -331,28 +349,37 @@ void OBSBasicStatusBar::UpdateDroppedFrames()
 	if (avgCongestion > 1.0f)
 		avgCongestion = 1.0f;
 
-	if (avgCongestion < EPSILON) {
-		statusSquare->setPixmap(greenPixmap);
-	} else if (fabsf(avgCongestion - 1.0f) < EPSILON) {
-		statusSquare->setPixmap(redPixmap);
-	} else {
-		QPixmap pixmap(20, 20);
+	lastCongestion = congestion;
 
-		float red = avgCongestion * 2.0f;
-		if (red > 1.0f)
-			red = 1.0f;
-		red *= 255.0;
+	if (disconnected)
+		return;
 
-		float green = (1.0f - avgCongestion) * 2.0f;
-		if (green > 1.0f)
-			green = 1.0f;
-		green *= 255.0;
+	bool update = firstCongestionUpdate;
+	float congestionOverTime = avgCongestion;
 
-		pixmap.fill(QColor(int(red), int(green), 0));
-		statusSquare->setPixmap(pixmap);
+	if (congestionArray.size() >= CONGESTION_UPDATE_SECONDS) {
+		congestionOverTime = accumulate(congestionArray.begin(),
+						congestionArray.end(), 0.0f) /
+				     (float)congestionArray.size();
+		congestionArray.clear();
+		update = true;
+	} else {
+		congestionArray.emplace_back(avgCongestion);
 	}
```

Comment:
I understand the desire to avoid rapid state changes, but taking up to 10 seconds for the UI to notify about connection issues feels way too long. Congestion typically comes in bursts, so a user will see dropped frames increasing but the congestion icon will still show an excellent connection until the next update which doesn't seem right.

Reply:
Updated it to 4 seconds, which seems more reasonable.